faunadb-java/src/test/java/com/faunadb/client/ClientSpec.java

```diff
@@ -3448,6 +3462,53 @@ public void onComplete() {
     capturedEvents.get();
   }
 
+  @Test
+  public void ParallelsQueriesTest() throws ExecutionException, InterruptedException {
+    Collection<Exception> syncCollection = Collections.synchronizedCollection(new ArrayList<>());
+    List<FaunaClient> clientPool = getClientPool();
+    Random rand = new Random();
+
+    ExecutorService executor = Executors.newFixedThreadPool(MAX_PARALLEL_ATTEMPT);
+    List<CompletableFuture<Optional<Exception>>> results = new ArrayList<>();
+    for (int i = 0; i < IN_PARALLEL_VALUE ; i++) {
+        int randomNum =  rand.nextInt((MAX_PARALLEL_ATTEMPT-1));
+        FaunaClient client = clientPool.get(randomNum);
+        CompletableFuture<Optional<Exception>> future
+                = CompletableFuture.supplyAsync(
+                () -> {
+                  try {
+                    Expr values = Arr(IntStream.rangeClosed(1, 10).mapToObj(Language::Value).collect(Collectors.toList()));
+                    client.query(Map(Paginate(Documents(Collection(PARALLEL_COLLECTION_NAME))), reference -> Get(reference))).get();
```

Comment:
I think this lambda `reference -> Get(reference)` can be simplified with `::` notation